proto/regen/ecocredit/v1/tx.proto

```diff
@@ -215,19 +224,111 @@ message MsgMintBatchCredits {
+  }
+
+  // issuance are the credits proposed by the user to be issued in the batch.
+  repeated BatchIssuance issuance = 4;
+
+  // A reference to a transaction or an event referencing the transaction
+  // which caused the transfer from other chain or registry.
+  // If not set (null), it means that the batch originate in Regen Ledger
+  // (doesn't come from external registry).
+  OriginTx origin_tx = 7;
+
+  // reference note for accounting, will be passed to an event
+  string note = 8;
+
+  message PendingBatchInfo {
+
+    // project is the project_id of the project this batch belongs to, or a
+    // new set of ProjectInfo to be used for the new project, if the project
+    // does not exist yet
+    oneof project {
+      string project_id = 1;
+      PendingProjectInfo project_info = 2;
+    }
+
+    // metadata is any arbitrary metadata attached to the credit batch.
+    string metadata = 3;
+
+    // start_date is the beginning of the period during which this credit batch
+    // was quantified and verified.
+    google.protobuf.Timestamp start_date = 4 [ (gogoproto.stdtime) = true ];
+
+    // end_date is the end of the period during which this credit batch was
+    // quantified and verified.
+    google.protobuf.Timestamp end_date = 5 [ (gogoproto.stdtime) = true ];
+
+    // If open is true we will enable future minting.
+    // Otherwise we will seal the batch and disable the future minting.
+    bool open = 6;
+  }
+
+  message PendingProjectInfo {
+
+    // class_id is the unique ID of the class within which the project is created.
+    string class_id = 2;
+
+    // metadata is any arbitrary metadata attached to the project.
+    string metadata = 3;
+
+    // project_location is the location of the project backing the credits in this
+    // batch. It is a string of the form
+    // <country-code>[-<sub-national-code>[ <postal-code>]], with the first two
+    // fields conforming to ISO 3166-2, and postal-code being up to 64
+    // alphanumeric characters. country-code is required, while sub-national-code
+    // and postal-code can be added for increasing precision.
+    string project_location = 4;
+
+    // project_id is an optional user-specified project ID which can be used
+    // instead of an auto-generated ID. If project_id is provided, it must be
+    // unique within the credit class and match the regex [A-Za-z0-9]{2,16}
+    // or else the operation will fail. If project_id is omitted an ID will
+    // automatically be generated.
+    string project_id = 5;
+  }
+}
+
+// MsgMintPendingBatchCreditsResponse is the Msg/MintPendingBatchCredits response type.
+message MsgMintPendingBatchCreditsResponse {
+  // batch_denom is the unique denomination ID of the newly created batch.
+  string batch_denom = 1;
+}
+
+// MsgApproveBatchCredits is the Msg/ApproveBatchCredits request type.
+message MsgApproveBatchCredits {
+  string batch_denom = 1;
+  repeated uint64 pending_issuance_id = 2;
```

Comment:
What is the `pending_issuance_id`? Did we add in an `issuance_id` field with dynamic minting of batches?